src/main/java/uk/gov/ch/oauth/tokens/SessionSignInModifier.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.ch.oauth.tokens;
+
+import java.util.Map;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.model.AccessToken;
+import uk.gov.companieshouse.session.model.UserProfile;
+
+public class SessionSignInModifier {
+
+    @SuppressWarnings("unchecked")
+    // This is necessary as the original data on the session is untyped, but expected to be of the correct types
+    static Map<String, Object> updateSignIn(final Object sInf, final Object sio) {
+        final Map<String, Object> original = (Map<String, Object>) sInf;
+        final Map<String, Object> extras = (Map<String, Object>) sio;
+        original.putAll(extras);
+        return original;
+    }
```

Comment:
The original method from Oauth2 can be removed now

Reply:
Good point